model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/solve_nonhydro.py

```diff
@@ -1429,57 +1420,28 @@ def run_corrector_step(
 
         log.debug("exchanging prognostic field 'vn'")
         self._exchange.exchange_and_wait(dims.EdgeDim, (prognostic_states.next.vn))
-        log.debug("corrector: start stencil 31")
-        self._compute_avg_vn(
+
+        self._compute_averaged_vn_and_fluxes_and_prepare_tracer_advection(
+            spatially_averaged_vn=self.z_vn_avg,
+            mass_flux_at_edges_on_model_levels=diagnostic_state_nh.mass_flux_at_edges_on_model_levels,
+            theta_v_flux_at_edges_on_model_levels=self.theta_v_flux_at_edges_on_model_levels,
+            substep_and_spatially_averaged_vn=prep_adv.vn_traj,
+            substep_averaged_mass_flux=prep_adv.mass_flx_me,
             e_flx_avg=self._interpolation_state.e_flx_avg,
             vn=prognostic_states.next.vn,
-            z_vn_avg=self.z_vn_avg,
-            horizontal_start=self._start_edge_lateral_boundary_level_5,
-            horizontal_end=self._end_edge_halo_level_2,
-            vertical_start=0,
-            vertical_end=self._grid.num_levels,
-            offset_provider=self._grid.connectivities,
-        )
-
-        log.debug("corrector: start stencil 32")
-        self._compute_mass_flux(
-            z_rho_e=z_fields.rho_at_edges_on_model_levels,
-            z_vn_avg=self.z_vn_avg,
+            rho_at_edges_on_model_levels=z_fields.rho_at_edges_on_model_levels,
             ddqz_z_full_e=self._metric_state_nonhydro.ddqz_z_full_e,
-            z_theta_v_e=z_fields.theta_v_at_edges_on_model_levels,
-            mass_fl_e=diagnostic_state_nh.mass_flux_at_edges_on_model_levels,
-            z_theta_v_fl_e=self.theta_v_flux_at_edges_on_model_levels,
+            theta_v_at_edges_on_model_levels=z_fields.theta_v_at_edges_on_model_levels,
+            prepare_advection=lprep_adv,
+            at_first_substep=at_first_substep,
+            r_nsubsteps=r_nsubsteps,
```

Comment:
can we put these 3 in the static args?

Reply:
I discussed with @OngChia and @yiluchen1066 , we think that `lprep_adv` can definitely go into the config, the other 2 change from stubstep to substep or change for high CFL.
I would like to put `lprep_adv` into the config in a separate PR though, since this will also change icon-exclaim.